### PR TITLE
[GH-15620] Migrated string refs in signup_email

### PR DESCRIPTION
**Pull request by varunks99**
#### Summary
This pull request migrates all string-refs in _./components/signup/signupemail/signupemail.jsx_ to functional ones using the `createRef` API.

#### Ticket Link
This fixes mattermost/mattermost-server#15620.

### CODE REVIEW

**Review by reflog — CHANGES_REQUESTED** [review]
@varunks99 - thanks for you contribution!
Please run `npm run fix` to address linting issues.

File: components/signup/signup_email/signup_email.jsx [hunk 1]
```diff
@@ -57,14 +57,18 @@ export default class SignupEmail extends React.PureComponent {
                 inviteId,
             };
         }
+
+        this.email = React.createRef();
+        this.name = React.createRef();
+        this.password = React.createRef();
```
[the remark this answers is not in the document]
Comment (varunks99) [reply components/signup/signup_email/signup_email.jsx]: @reflog, I renamed the refs and fixed the linting issues.

**Review by reflog — APPROVED** [review]
:100: Thanks, looking good!

**Review by Willyfrog — APPROVED** [review]
LGTM

**Review by jgilliam17 — APPROVED** [review]
Thank you @varunks99 
Tested, looks good to merge.
- Verified email sign-up works es expected.

**Comment by jfrerich** [comment]
@reflog, If this looks good to you, would you mind merging?  Thanks!

**Comment by reflog** [comment]
> @reflog, If this looks good to you, would you mind merging?  Thanks! 

Done

**Pull request merged**